src/data_model/transformations.jl

```diff
@@ -472,3 +472,917 @@ function apply_voltage_angle_difference_bounds!(eng::Dict{String,<:Any}, vad::Re
+    for prop in ["vm", "va", "vr_start", "vi_start", "vm_start", "va_start"]
+        if haskey(bus, prop)
+            vals = bus[prop][old_idx[t_to]]
+            @assert vals.==vals[1] "Cannot merge bus property $prop because the merged terminals differ."
+            bus[prop] = [bus[prop][findfirst(old_terminals.==t)] for t in merged_terminals]
+        end
+    end
+
+    # merge grounding properties (i.e. equivalent impedance of parallel impedances to ground if multiple groundings)
+    if haskey(bus, "grounded")
+        grounded = bus["grounded"]
+        if t_fr in grounded
+            idxs_fr = findall(grounded.==t_fr)
+            idxs_to = findall(grounded.==t_to)
+            idxs = [idxs_fr..., idxs_to...]
+            zgs = [bus["rg"][i]+im*bus["xg"][i] for i in idxs]
+            zg = any(iszero.(zgs)) ? im*0.0 : 1/sum(1/x for x in zgs)
+
+            idxs_other = setdiff(1:length(grounded), idxs)
+            bus["grounded"] = [grounded[idxs_other]..., t_to]
+            bus["rg"] = [bus["rg"][idxs_other]..., real.(zg)]
+            bus["xg"] = [bus["xg"][idxs_other]..., imag.(zg)]
+        end
+    end
+
+    #TODO take care of other bounds in the bus data model
+
+    # update connection properties to reflect the merge
+    for type in setdiff(intersect(pmd_eng_asset_types, keys(data_eng)), ["bus"])
+        for (_,comp) in data_eng[type]
+            # one-port components
+            if _is_oneport_component(comp) && comp["bus"]==bus_id
+                comp["connections"] = [t==t_fr ? t_to : t for t in comp["connections"]]
+            # two-port components
+            elseif _is_twoport_component(comp)
+                if comp["f_bus"]==bus_id
+                    comp["f_connections"] = [t==t_fr ? t_to : t for t in comp["f_connections"]]
+                end
+                if comp["t_bus"]==bus_id
+                    comp["t_connections"] = [t==t_fr ? t_to : t for t in comp["t_connections"]]
+                end
+            # multi-port components
+            elseif _is_multiport_component(comp)
+                for (w,bus_w) in enumerate(comp_bus)
+                    if bus_w==bus_id
+                        comp["connections"][w] = [t==t_fr ? t_to : t for t in comp["connections"][w]]
+                    end
+                end
+            end
+        end
+    end
+end
+
+
+""""
+Transform line loops (connected to a single bus), which are not allowed in the mathematical model.
+Lossy line loops are converted to equivalent shunts, and lossless ones (i.e. short-circuits) are represented by merging the short-circuited terminals. 
+The argument 'zero_series_impedance_threshold' controls the threshold below which the series impedance is considered to be a short-ciruit.
+This is useful because OpenDSS modelers have to insert tiny impedances to represent short-circuit reactors. 
+The addmittance to ground should be zero to trigger the short-circuit handling. 
+"""
```

Comment:
For exported functions, we should follow the new docstring pattern where the function is given (indented) first, see, e.g. transform_data_model function as an example.